packages/app/api-common/src/paginationUtils.ts

```diff
@@ -201,3 +201,22 @@ export async function getPaginatedEntriesByHasMore<T extends Record<string, unkn
 
   return resultArray
 }
+
+export async function traversePaginatedSourceByHasMore<
+  T extends Record<string, unknown>,
+>(
+  pagination: OptionalPaginationParams,
+  apiCall: (params: OptionalPaginationParams) => Promise<PaginatedResponse<T>>,
+  processPage: (pageData: PaginatedResponse<T>['data']) => Promise<void>,
+): Promise<void> {
```

Comment:
🟢 Could be interesting to be able to return the results of the processing step? something like
```
export async function traversePaginatedSourceByHasMore<
  T extends Record<string, unknown>,
  ReturnType = void,
>(
  pagination: OptionalPaginationParams,
  apiCall: (params: OptionalPaginationParams) => Promise<PaginatedResponse<T>>,
  processPage: (pageData: PaginatedResponse<T>['data']) => Promise<ReturnType> | ReturnType,
): Promise<ReturnType> {
```
An example usage could be to insert something on DB and return all new IDs. What do you think?